airbyte-integrations/connectors/source-google-analytics-data-api/integration_tests/configured_catalog.json

```diff
@@ -91,30 +98,33 @@
         "supported_sync_modes": ["full_refresh", "incremental"],
         "source_defined_cursor": true,
         "default_cursor_field": ["date"],
-        "source_defined_primary_key": [["uuid"]]
+        "source_defined_primary_key": [["property_id"], ["date"]]
       },
-      "sync_mode": "full_refresh",
-      "destination_sync_mode": "overwrite"
+      "sync_mode": "incremental",
+      "destination_sync_mode": "overwrite",
+      "primary_key": [["property_id"], ["date"]]
     },
     {
       "stream": {
         "name": "cohort_report",
         "json_schema": {},
         "supported_sync_modes": ["full_refresh"],
-        "source_defined_primary_key": [["uuid"]]
+        "source_defined_primary_key": [["property_id"], ["cohort"], ["cohortNthDay"]]
       },
       "sync_mode": "full_refresh",
-      "destination_sync_mode": "overwrite"
+      "destination_sync_mode": "overwrite",
+      "primary_key": [["property_id"], ["cohort"], ["cohortNthDay"]]
     },
     {
       "stream": {
         "name": "pivot_report",
         "json_schema": {},
         "supported_sync_modes": ["full_refresh"],
-        "source_defined_primary_key": [["uuid"]]
+        "source_defined_primary_key": [["property_id"], ["browser"], ["country"], ["language"]]
```

Comment:
Do we really want to use the `browser` field for  `pk` ? Is there anything else suitable?

Reply:
As you know all streams in this connector consist of dimensions and metrics. I did not find anything better than using all dimension fields as multi-field primary key.

Reply:
do you have the sample of the "browser" value to check before merge this in, I just doubt that those are unique)

Reply:
Yes, good catch!

I underestimated the problem initially and my concept only worked for streams that contained `"date"` dimension.
If the stream missed `"date"` dimension - I have a problem.

I still believe that I can finish this proof of concept :)

Now I have new improvement !!!

if the stream is missed `"date"` dimension new field is added:

`"date_range":"2023-04-01/2023-04-30"`

this field is also part of the PK.